packages/app/src/vis-packs/core/utils.ts

```diff
@@ -95,12 +93,16 @@ export function getImageInteractions(keepRatio: boolean): InteractionInfo[] {
   return keepRatio ? BASE_INTERACTIONS : INTERACTIONS_WITH_AXIAL_ZOOM;
 }
 
+function isBoolArray(val: ArrayValue<NumericLikeType>): val is boolean[] {
+  return Array.isArray(val) && typeof val[0] === 'boolean';
+}
+
 export function toNumArray(arr: ArrayValue<NumericLikeType>): NumArray {
-  if (typeof arr[0] === 'boolean') {
```

Comment:
TypeScript still thinks that `arr` is `number[] | boolean[]` both inside and after this block, so I'm implementing a proper type guard. Since it's so specific (the parameter has type `ArrayValue<NumericLikeType>`), I chose to keep the type guard close by.